### ISSUE TITLE
In "/v2/entities/<id>/attrs/<attr_name>/value" without entity id or attribute name in URL

### ISSUE THREAD

**Issue by iariasleon** [telefonicaid/fiware-orion#1487]
The status code is wrong.
The error and description messages are missing.

## Create Request:
```
POST http://localhost:1026/v2/entities
Content-Type: application/json
Fiware-Service: test_update_error
Fiware-ServicePath: /test
{"temperature": {"type": "celcius", "value": "34"}, "type": "house", "id": "room"}
```

## Update request without entity id in url:
```
 PUT http://qa-orion-fe-02:1026/v2/entities//attrs/speed/value
 Content-Type: application/json
 Fiware-Service: test_update_error
 Fiware-ServicePath: /test
 {"value": "80"}
```

#### Update response:
```
http code: 404
content-length: 0
```

#### Expected:
```
http code: 400 Bad Request 
{
    "error": "BadRequest",     
    "description": "service not found"
}
```

## Update request without attribute name in url:
```
 PUT http://qa-orion-fe-02:1026/v2/entities/room/attrs//value
 Content-Type: application/json
 Fiware-Service: test_update_error
Fiware-ServicePath: /test
 {"value": "80"}
```

#### Update response:
```
http code: 472
content-length: 0
```

#### Expected:
```
http code: 400 Bad Request 
{
    "error": "BadRequest",     
    "description": "service not found"
}
```

**Comment by fgalan** [comment]
Let's keep this issue *on hold*. Probably entity id "" will be unallowed at the end.

**Comment by fgalan** [comment]
Asigned to @iariasleon for a new test round.

**Comment by iariasleon** [comment]
The behavior is the same, still fails the test, tested in:
```
  "version" : "1.0.0-next",
  "git_hash" : "ee57ad7538026fd965184fa027194d9aa3a00ec6"
```

**Comment by iariasleon** [comment]
The behavior is the same, still fails the test, tested in:
```
  "version" : "1.0.0-next",
  "git_hash" : "ee57ad7538026fd965184fa027194d9aa3a00ec6"
```

**Comment by kzangeli** [comment]
This specific error is fixed in PR #2297, however, we accept wildcards in many other operations and we should implement the very same 'check for empty' in all of them.
As we aren't 100% sure on the exact error we want to return, the fixing of all of these cases will have to be put on hold until we decide the exact error to return.

**Comment by fgalan** [comment]
I'm not sure if this is a "wildcard" issue... The issue is related with what to do when an "structural URL field" is missing in the URL and this ends with `//`. Should be `//` interpreted as `/` (in which case the error is "service not found")? Or should be interpreted as "field is the empty string" (in which case the error is due to violation of id syntax rules in NGSIv2)? I think that is the point at the end.

**Comment by jmcanterafonseca** [comment]
I would prefer the latter , URIs should have single slash for separating components so a double slash means that something is wrong with the client app and it should be raised an error

**Comment by fgalan** [comment]
After discussion, we would opt for:

> Or should be interpreted as "field is the empty string" (in which case the error is due to violation of id syntax rules in NGSIv2)

**Comment by fgalan** [comment]
Taking into account the previous comment, the expected output will actually be:

```
http code: 400 Bad Request 
{
    "error": "BadRequest",     
    "description": "entity type length: 0, min length supported: 1"
}
```

@iariasleon pls adjust your test in that way

**Comment by fgalan** [comment]
...or, in the case of missing attribute:

```
http code: 400 Bad Request 
{
    "error": "BadRequest",     
    "description": "attribute name length: 0, min length supported: 1"
}
```

**Comment by fgalan** [comment]
Fixed by PR #2314. @iariasleon please have a look and close if you find it solved.

**Comment by iariasleon** [comment]
LGTM
```
http code: 400 - Bad Request
headers:
   date: Thu, 21 Jul 2016 08:46:39 GMT
   fiware-correlator: a443d17c-4f1f-11e6-8018-005056a20feb
   connection: Keep-Alive
   content-type: application/json
   content-length: 88
payload: {"error":"BadRequest","description":"attribute name length: 0, min length supported: 1"}
```
